pkg/gardenlet/controller/backupbucket/reconciler.go

```diff
@@ -296,7 +296,7 @@ func (r *Reconciler) deleteBackupBucket(
 		if !apierrors.IsNotFound(err) {
 			return reconcile.Result{}, err
 		}
-	} else {
```

Comment:
I am slightly confused by this change. Is this necessary at all?
The `if` block checks for `err != nil`. Hence, the `else` has to have `err == nil`.
My understanding was that the idiomatic way to write this is
```
if ...; err != nil {

} else {

}
```
and not
```
if ...; err != nil {

} else if err == nil {

}
```

What am I missing?

Reply:
Without this, the linter complains that we should exit early here, i.e., it wants us to do this:

```golang
if err := r.SeedClient.Get(seedCtx, client.ObjectKeyFromObject(extensionBackupBucket), extensionBackupBucket); err == nil {
	if lastError := extensionBackupBucket.Status.LastError; lastError != nil {
		r.Recorder.Event(backupBucket, corev1.EventTypeWarning, gardencorev1beta1.EventDeleteError, lastError.Description)

		if updateErr := r.updateBackupBucketStatusError(gardenCtx, backupBucket, lastError.Description+" Operation will be retried.", lastError); updateErr != nil {
			return reconcile.Result{}, fmt.Errorf("could not update status after deletion error: %w", updateErr)
		}
		return reconcile.Result{}, errors.New(lastError.Description)
	}
	log.Info("Extension BackupBucket not yet deleted", "extensionBackupBucket", client.ObjectKeyFromObject(extensionBackupBucket))
	return reconcile.Result{RequeueAfter: RequeueDurationWhenResourceDeletionStillPresent}, nil
}

if !apierrors.IsNotFound(err) {
	return reconcile.Result{}, err
}
```

However, I don't like this/think this is good style. Hence, I added this `else if` to make it happy. Alternatively, we can annotate the block with

```golang
//nolint:revive
```

and remove the `else if` again. Personally, I like what I did slightly more than simply tell it to not lint at all.

Reply:
Ok, thanks for explaining. I was surprised by this change, but with the explanation I agree that the other two options are not great either. Hence, the current approach is fine with me.